packages/Teams/TeamsBotComponent.cs

```diff
@@ -15,6 +16,22 @@ public class TeamsBotComponent : BotComponent
         /// <inheritdoc/>
         public override void ConfigureServices(IServiceCollection services, IConfiguration configuration)
         {
+            if (services == null)
+            {
+                throw new ArgumentNullException(nameof(services));
+            }
+
+            if (configuration == null)
+            {
+                throw new ArgumentNullException(nameof(configuration));
+            }
+
+            var ettings = configuration.GetSection(ComponentSettings.SettingsKey).Get<ComponentSettings>() ?? new ComponentSettings();
+            if (ettings.SSOMiddleware)
+            {
+                services.AddSingleton<IMiddleware, SSOTokenExchangeMiddleware>();
```

Comment:
This will be picked up by the adapter.

SSOTokenExchangeMiddleware is a short-circuit middleware, so will affect processing of other middlewares after it in the stack.  This is a known issue with middleware.